examples/seismic/model.py

```diff
@@ -437,58 +438,63 @@ def critical_dt(self):
         coeff = 0.38 if len(self.shape) == 3 else 0.42
         return self.dtype(coeff * np.min(self.spacing) / (self.scale*np.max(self.vp)))
 
+    def get_vp(self, exclude_pml=True):
+        """Returns the velocity model
+           :param exclude_pml : excludes the pml region if true.
+        """
+        if exclude_pml:
+            dim = len(self.shape)
+            if dim == 1:
+                return np.sqrt(1. / self.m.data[self.nbpml:-self.nbpml])
```

Comment:
I'm 99.9% sure there's a way of doing this whole block ``if dim == 1 ... elif ... else `` with a single numpy line, ie, slicing parametrically in the number of dimensions